src/main/java/org/sunbird/trending/controller/TrendingServiceImpl.java

```diff
@@ -178,84 +179,72 @@ private String getEndDateFormat() {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Filter is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           List<String> contextTypeList = new ArrayList<>();
+           if (CollectionUtils.isNotEmpty(contextTypeList)) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "ContextType is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           String org = (filter).get(ORGANISATION) == null ? "" : ((String) (filter).get(ORGANISATION));
+           if (org.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Organization is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           Map<String, String> payloadToRedisKeyMapping = serverProperties.getPayloadToRedisKeyMapping();
+           contextTypeList.addAll(payloadToRedisKeyMapping.keySet());
+           Map<String, Object> aggregateData = new HashMap<>();
+           List<String> topContentRedis;
+           List<String> limitCourses;
+           int limit = Optional.ofNullable(request.get(LIMIT)).map(l -> (Integer) l).orElse(0);
+           for (String contextType : contextTypeList) {
+               String contextTypeValue = payloadToRedisKeyMapping.get(contextType);
+               topContentRedis = redisCacheMgr.hget(contextTypeValue, serverProperties.getRedisInsightIndex(), new String[]{org});
+               limitCourses = this.fetchIds(topContentRedis.get(0), limit, "");
+               if (CollectionUtils.isNotEmpty(limitCourses) && null != limitCourses.get(0)) {
+                   aggregateData.put(contextType, limitCourses);
+               }
+           }
+           Map<String, Object> compositeSearchRes;
+           List<Map<String, Object>> contentList = new ArrayList<>();
+           Map<String, Object> resultMap;
+           if (!aggregateData.isEmpty()) {
+               List<String> compositeKeyList = new ArrayList<>(payloadToRedisKeyMapping.keySet());
+               for (int i = 0; i < aggregateData.size(); i++) {
+                   List<String> searchIds = (List<String>) aggregateData.get(compositeKeyList.get(i));
+                   compositeSearchRes = compositeSearch(searchIds, token);
+                   if (null == compositeSearchRes)
+                       compositeSearchRes = new HashMap<>();
+                   resultMap = compositeSearchRes.get(RESULT) == null ? new HashMap<>() : (Map<String, Object>) compositeSearchRes.get(RESULT);
+                   contentList = resultMap.get(CONTENT) == null ? new ArrayList<>() : (List<Map<String, Object>>) resultMap.get(CONTENT);
+               }
+           }
+           response.put(RESPONSE, contentList);
```

Comment:
It will only contain the content list of the last contextType because within the loop above, contentList is consistently being replaced.

Reply:
Implimented as suggested